File: jdevkit/src/test/java/com/github/klee0kai/android_devkit/threads/ThreadTests.java
package com.github.klee0kai.android_devkit.threads;

import static org.junit.Assert.assertEquals;

import org.junit.Test;

import java.util.LinkedList;
import java.util.concurrent.ExecutionException;
import java.util.concurrent.Future;
import java.util.concurrent.ThreadPoolExecutor;

public class ThreadTests {

    private static final int RUN_COUNT = 300;

    @Test(timeout = 100)
    public void sequence_run() throws ExecutionException, InterruptedException {
        //Give
        LinkedList<Integer> runs = new LinkedList<>();
        ThreadPoolExecutor secThread = Threads.newSingleThreadExecutor("secThread");

        //When
        Future lastFuture = null;
        for (int i = 0; i < RUN_COUNT; i++) {
            int finalI = i;
            lastFuture = secThread.submit(() -> {
                runs.add(finalI);
            });
        }
        lastFuture.get();

        //then
        for (int i = 0; i < RUN_COUNT; i++) {
            assertEquals(i, runs.get(i).intValue());
        }

    }
}

File: kotlin_hummus/src/test/kotlin/com/github/klee0kai/hummus/threads/ThreadTests.kt
package com.github.klee0kai.hummus.threads

import com.github.klee0kai.hummus.threads.Threads
import org.junit.Assert
import org.junit.Test
import java.util.*
import java.util.concurrent.ExecutionException
import java.util.concurrent.Future
import java.util.concurrent.ThreadPoolExecutor

class ThreadTests {

    private val RUN_COUNT = 300

    @Test
    @Throws(InterruptedException::class, ExecutionException::class)
    fun sequence_run() {
        //Give
        val runs = LinkedList<Int>()
        val secThread: ThreadPoolExecutor = Threads.newSingleThreadExecutor("secThread")

        //When
        var lastFuture: Future<*>? = null
        for (i in 0 until RUN_COUNT) {
            lastFuture = secThread.submit { runs.add(i) }
        }
        lastFuture!!.get()

        //then
        for (i in 0 until RUN_COUNT) {
            Assert.assertEquals(i.toLong(), runs[i].toLong())
        }
    }
}